### PR TITLE
Memory leaks and performance fixes

### PR DESCRIPTION
**Pull request by gaurav1086**
Memory leaks and some performance fixes

aten/src/ATen/core/ivalue.cpp: 77 and 91 : Function parameter 'start' and 'finish' should be passed by reference instead of value.

aten/src/ATen/native/Convolution.cpp:472: Removed Unused variable: output_ss

aten/src/ATen/native/Normalization.cpp:460 : Variable 'use_cudnn' declared/defined together

aten/src/TH/THMemoryFile.cpp:580: initialize posEol, i = 0

aten/src/TH/vector/VSX.cpp: Memory leak: y multiple locations

aten/src/THC/THCCachingHostAllocator.cpp:276: Unsigned variable 'size' can't be negative so it is unnecessary to test it.

### CODE REVIEW

**Comment by gaurav1086** [comment]
Review requested. Thank you.

**Comment by gaurav1086** [comment]
@ebetica @yf225 @goldsborough please review. Thank you.

File: aten/src/ATen/native/RNN.cpp [hunk 1]
```diff
@@ -293,10 +293,10 @@ static std::vector<QuantizedCellParamsDynamic> gather_quantized_params_dynamic(
     auto bias_hh = packed_struct_hh.bias.value_or(undefined);
     result.emplace_back(params[i], params[i + 1], bias_ih, bias_hh);
   }
-  return result;
 #else // USE_FBGEMM
   TORCH_INTERNAL_ASSERT(false, "Tried to use quantized RNN wihtout FBGEMM!")
 #endif // USE_FBGEMM
+  return result;
```
Comment (jamesr66a) [inline aten/src/ATen/native/RNN.cpp]: What's the reason for this change? Is this to workaround some no-return-statement lint issue?

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 7e020bdc:


* **9/9** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=7e020bdc64b2bd20eb9393217e69f7a73220815d).

### :detective: 9 new failures recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4287488) caffe2_onnx_py2_gcc5_ubuntu16_04_build (1/9)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87541907) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87541907))
<details>
<summary>
<code>Jan 17 02:01:21 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:10: error: 'TensorList' is not a member of 'c10::IValue::Tag'</code>
</summary>

```
Jan 17 02:01:21 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp: In function 'std::ostream& c10::operator<<(std::ostream&, const c10::IValue&)': 
Jan 17 02:01:21 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:151:10: error: 'IntList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:01:21      case IValue::Tag::IntList: 
Jan 17 02:01:21           ^ 
Jan 17 02:01:21 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:153:10: error: 'DoubleList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:01:21      case IValue::Tag::DoubleList: 
Jan 17 02:01:21           ^ 
Jan 17 02:01:21 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:155:10: error: 'BoolList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:01:21      case IValue::Tag::BoolList: 
Jan 17 02:01:21           ^ 
Jan 17 02:01:21 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:10: error: 'TensorList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:01:21      case IValue::Tag::TensorList: 
Jan 17 02:01:21           ^ 
Jan 17 02:01:21 caffe2/CMakeFiles/torch_cpu.dir/build.make:1090: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o' failed 
Jan 17 02:01:21 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o] Error 1 
Jan 17 02:01:21 make[2]: *** Waiting for unfinished jobs.... 
Jan 17 02:01:21 In file included from /var/lib/jenkins/workspace/build/aten/src/ATen/core/TensorBody.h:5:0, 
Jan 17 02:01:21                  from /var/lib/jenkins/workspace/aten/src/ATen/Tensor.h:11, 
Jan 17 02:01:21                  from /var/lib/jenkins/workspace/aten/src/ATen/Context.h:4, 
Jan 17 02:01:21                  from /var/lib/jenkins/workspace/aten/src/ATen/ATen.h:5, 
Jan 17 02:01:21                  from /var/lib/jenkins/workspace/aten/src/ATen/native/ReflectionPad.cpp:1: 
```

</details>

#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4287485) caffe2_onnx_py3_6_clang7_ubuntu16_04_build (2/9)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87542089) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87542089))
<details>
<summary>
<code>Jan 17 02:01:52 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:23: error: no member named 'TensorList' in 'c10::IValue::Tag'</code>
</summary>

```
Jan 17 02:01:52          ~~~~~^ 
Jan 17 02:01:52 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:151:23: error: no member named 'IntList' in 'c10::IValue::Tag' 
Jan 17 02:01:52     case IValue::Tag::IntList: 
Jan 17 02:01:52          ~~~~~~~~~~~~~^ 
Jan 17 02:01:52 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:153:23: error: no member named 'DoubleList' in 'c10::IValue::Tag' 
Jan 17 02:01:52     case IValue::Tag::DoubleList: 
Jan 17 02:01:52          ~~~~~~~~~~~~~^ 
Jan 17 02:01:52 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:155:23: error: no member named 'BoolList' in 'c10::IValue::Tag' 
Jan 17 02:01:52     case IValue::Tag::BoolList: 
Jan 17 02:01:52          ~~~~~~~~~~~~~^ 
Jan 17 02:01:52 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:23: error: no member named 'TensorList' in 'c10::IValue::Tag' 
Jan 17 02:01:52     case IValue::Tag::TensorList: 
Jan 17 02:01:52          ~~~~~~~~~~~~~^ 
Jan 17 02:01:52 8 errors generated. 
Jan 17 02:01:52 caffe2/CMakeFiles/torch_cpu.dir/build.make:1090: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o' failed 
Jan 17 02:01:52 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o] Error 1 
Jan 17 02:01:52 make[2]: *** Waiting for unfinished jobs.... 
Jan 17 02:01:52 /var/lib/jenkins/workspace/aten/src/ATen/native/DilatedMaxPool3d.cpp:558:42: warning: 'get_contiguous_memory_format' is deprecated [-Wdeprecated-declarations] 
Jan 17 02:01:52   auto gradInput = at::zeros_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT); 
Jan 17 02:01:52                                          ^ 
Jan 17 02:01:52 /var/lib/jenkins/workspace/c10/core/MemoryFormat.h:32:46: note: expanded from macro 'LEGACY_CONTIGUOUS_MEMORY_FORMAT' 
```

</details>

#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4287493) pytorch_linux_xenial_py2_7_9_build (3/9)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87543090) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87543090))
<details>
<summary>
<code>Jan 17 02:02:23 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:23: error: 'TensorList' is not a member of 'c10::IValue::Tag'</code>
</summary>

```
Jan 17 02:02:23 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp: In function 'std::ostream& c10::operator<<(std::ostream&, const c10::IValue&)': 
Jan 17 02:02:23 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:151:23: error: 'IntList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:02:23      case IValue::Tag::IntList: 
Jan 17 02:02:23                        ^~~~~~~ 
Jan 17 02:02:23 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:153:23: error: 'DoubleList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:02:23      case IValue::Tag::DoubleList: 
Jan 17 02:02:23                        ^~~~~~~~~~ 
Jan 17 02:02:23 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:155:23: error: 'BoolList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:02:23      case IValue::Tag::BoolList: 
Jan 17 02:02:23                        ^~~~~~~~ 
Jan 17 02:02:23 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:23: error: 'TensorList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:02:23      case IValue::Tag::TensorList: 
Jan 17 02:02:23                        ^~~~~~~~~~ 
Jan 17 02:02:23 caffe2/CMakeFiles/torch_cpu.dir/build.make:1090: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o' failed 
Jan 17 02:02:23 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o] Error 1 
Jan 17 02:02:23 make[2]: *** Waiting for unfinished jobs.... 
Jan 17 02:02:24 In file included from /var/lib/jenkins/workspace/build/aten/src/ATen/core/TensorBody.h:5:0, 
Jan 17 02:02:24                  from /var/lib/jenkins/workspace/aten/src/ATen/Tensor.h:11, 
Jan 17 02:02:24                  from /var/lib/jenkins/workspace/aten/src/ATen/Context.h:4, 
Jan 17 02:02:24                  from /var/lib/jenkins/workspace/aten/src/ATen/ATen.h:5, 
Jan 17 02:02:24                  from /var/lib/jenkins/workspace/aten/src/ATen/native/AdaptiveAveragePooling.cpp:1: 
```

</details>

#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4287484) pytorch_linux_xenial_py3_clang5_asan_build (4/9)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87543123) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87543123))
<details>
<summary>
<code>Jan 17 02:02:31 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:23: error: no member named 'TensorList' in 'c10::IValue::Tag'</code>
</summary>

```
Jan 17 02:02:31               ^~~~~~~ 
Jan 17 02:02:31 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:151:23: error: no member named 'IntList' in 'c10::IValue::Tag' 
Jan 17 02:02:31     case IValue::Tag::IntList: 
Jan 17 02:02:31          ~~~~~~~~~~~~~^ 
Jan 17 02:02:31 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:153:23: error: no member named 'DoubleList' in 'c10::IValue::Tag' 
Jan 17 02:02:31     case IValue::Tag::DoubleList: 
Jan 17 02:02:31          ~~~~~~~~~~~~~^ 
Jan 17 02:02:31 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:155:23: error: no member named 'BoolList' in 'c10::IValue::Tag' 
Jan 17 02:02:31     case IValue::Tag::BoolList: 
Jan 17 02:02:31          ~~~~~~~~~~~~~^ 
Jan 17 02:02:31 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:23: error: no member named 'TensorList' in 'c10::IValue::Tag' 
Jan 17 02:02:31     case IValue::Tag::TensorList: 
Jan 17 02:02:31          ~~~~~~~~~~~~~^ 
Jan 17 02:02:31 10 errors generated. 
Jan 17 02:02:31 caffe2/CMakeFiles/torch_cpu.dir/build.make:1090: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o' failed 
Jan 17 02:02:31 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o] Error 1 
Jan 17 02:02:31 make[2]: *** Waiting for unfinished jobs.... 
Jan 17 02:02:32 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Jan 17 02:02:32 CMakeFiles/Makefile2:5179: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Jan 17 02:02:32 make: *** [all] Error 2 
Jan 17 02:02:32 Makefile:138: recipe for target 'all' failed 
```

</details>

#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4287489) binary_linux_libtorch_2_7m_cpu_devtoolset7_shared-with-deps_build (5/9)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87543172) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87543172))
<details>
<summary>
<code>Jan 17 02:02:33 /pytorch/aten/src/ATen/core/ivalue.cpp:159:23: error: ‘TensorList’ is not a member of ‘c10::IValue::Tag’</code>
</summary>

```
Jan 17 02:02:33 /pytorch/aten/src/ATen/core/ivalue.cpp: In function ‘std::ostream& c10::operator<<(std::ostream&, const c10::IValue&)’: 
Jan 17 02:02:33 /pytorch/aten/src/ATen/core/ivalue.cpp:151:23: error: ‘IntList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:33      case IValue::Tag::IntList: 
Jan 17 02:02:33                        ^~~~~~~ 
Jan 17 02:02:33 /pytorch/aten/src/ATen/core/ivalue.cpp:153:23: error: ‘DoubleList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:33      case IValue::Tag::DoubleList: 
Jan 17 02:02:33                        ^~~~~~~~~~ 
Jan 17 02:02:33 /pytorch/aten/src/ATen/core/ivalue.cpp:155:23: error: ‘BoolList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:33      case IValue::Tag::BoolList: 
Jan 17 02:02:33                        ^~~~~~~~ 
Jan 17 02:02:33 /pytorch/aten/src/ATen/core/ivalue.cpp:159:23: error: ‘TensorList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:33      case IValue::Tag::TensorList: 
Jan 17 02:02:33                        ^~~~~~~~~~ 
Jan 17 02:02:33 [ 50%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/Activation.cpp.o 
Jan 17 02:02:33 [ 50%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AdaptiveAveragePooling.cpp.o 
Jan 17 02:02:34 [ 50%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AdaptiveAveragePooling3d.cpp.o 
Jan 17 02:02:34 [ 50%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AdaptiveMaxPooling2d.cpp.o 
Jan 17 02:02:34 [ 51%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AdaptiveMaxPooling3d.cpp.o 
Jan 17 02:02:34 [ 51%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AffineGridGenerator.cpp.o 
Jan 17 02:02:34 gmake[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o] Error 1 
Jan 17 02:02:34 gmake[2]: *** Waiting for unfinished jobs.... 
```

</details>

#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4287477) binary_linux_libtorch_2_7m_cpu_gcc5_4_cxx11-abi_shared-with-deps_build (6/9)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87543209) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87543209))
<details>
<summary>
<code>Jan 17 02:02:39 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o</code>
</summary>

```
Jan 17 02:02:37 [1635/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/interned_strings.cpp.o 
Jan 17 02:02:37 [1636/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/register_symbols.cpp.o 
Jan 17 02:02:37 [1637/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/LegacyDefinitions.cpp.o 
Jan 17 02:02:38 [1638/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/dispatch/OperatorEntry.cpp.o 
Jan 17 02:02:38 [1639/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/strtod.cpp.o 
Jan 17 02:02:38 [1640/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/script/error_report.cpp.o 
Jan 17 02:02:38 [1641/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/dispatch/Dispatcher.cpp.o 
Jan 17 02:02:38 [1642/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/function_schema.cpp.o 
Jan 17 02:02:39 [1643/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/op_registration/infer_schema.cpp.o 
Jan 17 02:02:39 [1644/3018] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o 
Jan 17 02:02:39 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o  
DTH_HAVE_THREAD -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -Wno-maybe-uninitialized -fvisibility=hidden -O2 -DCAFFE2_BUILD_MAIN_LIB -DASMJIT_STATIC -std=gnu++14 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o -MF caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o.d -o caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o -c ../aten/src/ATen/core/ivalue.cpp 
Jan 17 02:02:39 ../aten/src/ATen/core/ivalue.cpp: In member function ‘c10::TypePtr c10::IValue::type() const’: 
Jan 17 02:02:39 ../aten/src/ATen/core/ivalue.cpp:39:10: error: ‘IntList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:39      case Tag::IntList: 
Jan 17 02:02:39           ^ 
Jan 17 02:02:39 ../aten/src/ATen/core/ivalue.cpp:41:10: error: ‘DoubleList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:39      case Tag::DoubleList: 
Jan 17 02:02:39           ^ 
Jan 17 02:02:39 ../aten/src/ATen/core/ivalue.cpp:43:10: error: ‘BoolList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:39      case Tag::BoolList: 
```

</details>

#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4287480) pytorch_linux_xenial_py3_clang5_mobile_build (7/9)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87543265) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87543265))
<details>
<summary>
<code>Jan 17 02:02:46 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:10: error: 'TensorList' is not a member of 'c10::IValue::Tag'</code>
</summary>

```
Jan 17 02:02:46 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp: In function 'std::ostream& c10::operator<<(std::ostream&, const c10::IValue&)': 
Jan 17 02:02:46 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:151:10: error: 'IntList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:02:46      case IValue::Tag::IntList: 
Jan 17 02:02:46           ^ 
Jan 17 02:02:46 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:153:10: error: 'DoubleList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:02:46      case IValue::Tag::DoubleList: 
Jan 17 02:02:46           ^ 
Jan 17 02:02:46 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:155:10: error: 'BoolList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:02:46      case IValue::Tag::BoolList: 
Jan 17 02:02:46           ^ 
Jan 17 02:02:46 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:10: error: 'TensorList' is not a member of 'c10::IValue::Tag' 
Jan 17 02:02:46      case IValue::Tag::TensorList: 
Jan 17 02:02:46           ^ 
Jan 17 02:02:46 caffe2/CMakeFiles/torch_cpu.dir/build.make:1072: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o' failed 
Jan 17 02:02:46 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o] Error 1 
Jan 17 02:02:46 make[2]: *** Waiting for unfinished jobs.... 
Jan 17 02:02:46 In file included from /var/lib/jenkins/workspace/build_mobile/aten/src/ATen/core/TensorBody.h:5:0, 
Jan 17 02:02:46                  from /var/lib/jenkins/workspace/aten/src/ATen/Tensor.h:11, 
Jan 17 02:02:46                  from /var/lib/jenkins/workspace/aten/src/ATen/Context.h:4, 
Jan 17 02:02:46                  from /var/lib/jenkins/workspace/aten/src/ATen/ATen.h:5, 
Jan 17 02:02:46                  from /var/lib/jenkins/workspace/aten/src/ATen/native/LossNLL.cpp:1: 
```

</details>

#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4287474) binary_linux_manywheel_2_7mu_cpu_devtoolset7_build (8/9)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87543387) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87543387))
<details>
<summary>
<code>Jan 17 02:02:51 /pytorch/aten/src/ATen/core/ivalue.cpp:159:23: error: ‘TensorList’ is not a member of ‘c10::IValue::Tag’</code>
</summary>

```
Jan 17 02:02:51 /pytorch/aten/src/ATen/core/ivalue.cpp: In function ‘std::ostream& c10::operator<<(std::ostream&, const c10::IValue&)’: 
Jan 17 02:02:51 /pytorch/aten/src/ATen/core/ivalue.cpp:151:23: error: ‘IntList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:51      case IValue::Tag::IntList: 
Jan 17 02:02:51                        ^~~~~~~ 
Jan 17 02:02:51 /pytorch/aten/src/ATen/core/ivalue.cpp:153:23: error: ‘DoubleList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:51      case IValue::Tag::DoubleList: 
Jan 17 02:02:51                        ^~~~~~~~~~ 
Jan 17 02:02:51 /pytorch/aten/src/ATen/core/ivalue.cpp:155:23: error: ‘BoolList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:51      case IValue::Tag::BoolList: 
Jan 17 02:02:51                        ^~~~~~~~ 
Jan 17 02:02:51 /pytorch/aten/src/ATen/core/ivalue.cpp:159:23: error: ‘TensorList’ is not a member of ‘c10::IValue::Tag’ 
Jan 17 02:02:51      case IValue::Tag::TensorList: 
Jan 17 02:02:51                        ^~~~~~~~~~ 
Jan 17 02:02:52 [ 50%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AdaptiveAveragePooling3d.cpp.o 
Jan 17 02:02:52 [ 50%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AdaptiveMaxPooling2d.cpp.o 
Jan 17 02:02:52 [ 50%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AdaptiveMaxPooling3d.cpp.o 
Jan 17 02:02:52 [ 50%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AffineGridGenerator.cpp.o 
Jan 17 02:02:53 [ 50%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/AveragePool2d.cpp.o 
Jan 17 02:02:53 gmake[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o] Error 1 
Jan 17 02:02:53 gmake[2]: *** Waiting for unfinished jobs.... 
Jan 17 02:02:54 In file included from /pytorch/build/aten/src/ATen/core/TensorBody.h:5:0, 
```

</details>

#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4287491) pytorch_xla_linux_xenial_py3_6_clang7_build (9/9)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87543481) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87543481))
<details>
<summary>
<code>Jan 17 02:03:19 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:23: error: no member named 'TensorList' in 'c10::IValue::Tag'</code>
</summary>

```
Jan 17 02:03:19          ~~~~~^ 
Jan 17 02:03:19 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:151:23: error: no member named 'IntList' in 'c10::IValue::Tag' 
Jan 17 02:03:19     case IValue::Tag::IntList: 
Jan 17 02:03:19          ~~~~~~~~~~~~~^ 
Jan 17 02:03:19 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:153:23: error: no member named 'DoubleList' in 'c10::IValue::Tag' 
Jan 17 02:03:19     case IValue::Tag::DoubleList: 
Jan 17 02:03:19          ~~~~~~~~~~~~~^ 
Jan 17 02:03:19 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:155:23: error: no member named 'BoolList' in 'c10::IValue::Tag' 
Jan 17 02:03:19     case IValue::Tag::BoolList: 
Jan 17 02:03:19          ~~~~~~~~~~~~~^ 
Jan 17 02:03:19 /var/lib/jenkins/workspace/aten/src/ATen/core/ivalue.cpp:159:23: error: no member named 'TensorList' in 'c10::IValue::Tag' 
Jan 17 02:03:19     case IValue::Tag::TensorList: 
Jan 17 02:03:19          ~~~~~~~~~~~~~^ 
Jan 17 02:03:19 8 errors generated. 
Jan 17 02:03:19 caffe2/CMakeFiles/torch_cpu.dir/build.make:1090: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o' failed 
Jan 17 02:03:19 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/core/ivalue.cpp.o] Error 1 
Jan 17 02:03:19 make[2]: *** Waiting for unfinished jobs.... 
Jan 17 02:03:19 CMakeFiles/Makefile2:5335: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Jan 17 02:03:19 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Jan 17 02:03:19 Makefile:138: recipe for target 'all' failed 
Jan 17 02:03:19 make: *** [all] Error 2 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

File: aten/src/ATen/native/RNN.cpp (hunk 1 above)
Comment (gaurav1086) [reply aten/src/ATen/native/RNN.cpp]: @jamesr66a , thanks for the review. I added that shut off the compiler since there has to be an unconditional return variable of type std::vector<QuantizedCellParamsDynamic> regardless of USE_FBGEMM macro.

**Comment by gaurav1086** [comment]
Closing this and creating a separate PR for memory leaks.
